src/pybind/mgr/dashboard_v2/HACKING.rst

```diff
@@ -508,3 +508,160 @@ The settings management implementation will make sure that if you change a
+
+
+How to get the list of executing and finished asynchronous tasks?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The list of executing and finished tasks is included in the ``Summary``
+controller, which is already polled every 5 seconds by the dashboard frontend.
+But we also provide a dedicated controller to get the same list of executing
+and finished tasks.
+
+The ``Task`` controller exposes the ``/api/task`` endpoint that returns the
+list of executing and finished tasks. This endpoint accepts the ``namespace``
+parameter that accepts a glob expression as its value.
+For instance, an HTTP GET request of the URL ``/api/task?namespace=rbd/*``
+will return all executing and finished tasks which namespace starts with
+``rbd/``.
+
+To prevent the finished tasks list from growing unboundly, the finished tasks
+will be maintained in memory for 1 minute. After a minute, when the finished
+task information is retrieved, either by the summary controller or by the task
+controller, it is automatically deleted from the list and it will not be
+included in further task queries.
+
+Each executing task is represented by the following dictionary::
+
+  {
+    'namespace': "namespace",  # str
+    'metadata': { },  # dict
+    'begin_time': 0.0,  # float
+    'progress': 0  # int (percentage)
+  }
+
+Each finished task is represented by the following dictionary::
+
+  {
+    'namespace': "namespace",  # str
+    'metadata': { },  # dict
+    'begin_time': 0.0,  # float
+    'end_time': 0.0,  # float
+    'latency': 0.0,  # float
+    'progress': 0  # int (percentage)
+    'success': True,  # bool
+    'ret_value': None,  # object, populated only if 'success' == True
+    'exception': None,  # str, populated only if 'success' == False
+  }
+
+
+How to updated the execution progress of an asynchronous task?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The asynchronous tasks infrastructure provides support for updating the
+execution progress of an executing task.
+The progress can be updated from within the code the task is executing, which
+usually is the place where we have the progress information available.
+
+To upgrade the progress from within the task code, the ``TaskManager`` class
+provides a method to retrieve the current task object::
+
+  TaskManager.current_task()
+
+The above method returns the current ``AsyncTask`` object. The ``AsyncTask``
+object provides two public methods to update the execution progress value: the
+``set_progress(percentage)``, and the ``inc_progress(delta)`` methods.
+
+The ``set_progress`` method receives as argument an integer value representing
+the absolute percentage that we want to set to the task.
+
+The ``inc_progress`` method receives as argument an integer value representing
+the delta we want to increment to the current execution progress percentage.
+
+Now we show a full example of a controller that triggers a new task and
```

Comment:
Instead of

`Now we show a full example of a controller [...]`

I'd go with

`Take the following example of a controler [...]`